src/frontend/src/pages/PIADetailPage/index.tsx

```diff
@@ -71,13 +92,60 @@ const PIADetailPage = () => {
         }
       }
     })();
-  }, [id, navigate, pia.hasAddedPiToDataElements, pia.ministry, win]);
+  }, [
+    id,
+    navigate,
+    pia.hasAddedPiToDataElements,
+    pia.ministry,
+    pia.status,
+    win,
+  ]);
 
   const handleAlertClose = () => {
     setFetchPiaError('');
     setDownloadError('');
   };
 
+  const handleEdit = () => {
+    if (isMPO && piaStatus !== 'INCOMPLETE') {
+      setModalConfirmLabel(messages.Modal.ConfirmLabel.en);
+      setModalCancelLabel(messages.Modal.CancelLabel.en);
+      setModalTitleText(messages.Modal.TitleText.en);
+      setModalParagraph(messages.Modal.ParagraphText.en);
+      setShowModal(true);
+    } else {
+      navigate(`${routes.PIA_INTAKE}/${id}`, {
+        state: pia,
+      });
+    }
+  };
+
+  const handleModalClose = async (event: any) => {
```

Comment:
should we be more specific with event type?

Reply:
done